app/controllers/api/v0/api_controller.rb

```diff
@@ -81,13 +81,12 @@ def authenticate_with_api_key_or_current_user!
         error_unauthorized unless user
       end
 
-      def add_missing_api_key_warning_header
+      def add_deprecation_warning_header
         return if headers["Accept"].present? &&
-          headers["Accept"].include?("application/vnd.forem.api-v#{@version}+json") &&
-          headers["api-key"].present?
+          headers["Accept"].include?("application/vnd.forem.api-v#{@version}+json")
 
         # rubocop:disable Layout/LineLength
-        response.headers["Warning"] = "299 - This endpoint will require the `api-key` header and the `Accept` header to be set to `application/vnd.forem.api-v1+json` in future."
+        response.headers["Warning"] = "299 - This endpoint is part of the V0 (beta) API. To start using the V1 endpoints add the `Accept` header and set it to `application/vnd.forem.api-v1+json`. Visit https://developers.forem.com/api for more information."
```

Comment:
Since we rolled back the api key decision previously made for V1 on some endpoints (see #18340) we need to update the logic that raises the deprecation warning. This is the most straightforward edit I could come up with. Totally open to copy edit 🙂

Reply:
As a bonus this will close https://github.com/forem/forem/issues/18240!